node/src/sub_lib/neighborhood.rs

```diff
@@ -478,157 +536,248 @@ mod tests {
     }
 
     #[test]
-    fn try_dismantle_works_for_ethereum_mainnet() {
-        let descriptor = "masq://eth.as45cs5c5:1.2.3.4:4444";
+    fn parse_works_for_ethereum_mainnet() {
+        let descriptor = "masq://eth-mainnet:as45cs5c5@1.2.3.4:4444";
 
-        let result = NodeDescriptor::try_dismantle_str(descriptor).unwrap();
+        let result = NodeDescriptor::parse_url(descriptor).unwrap();
 
-        assert_eq!(
-            result,
-            (Blockchain::EthMainnet, "as45cs5c5", "1.2.3.4:4444")
-        )
+        assert_eq!(result, (Chain::EthMainnet, "as45cs5c5", "1.2.3.4:4444"))
     }
 
     #[test]
-    fn try_dismantle_str_works_for_ethereum_testnet_ropsten() {
-        let descriptor = "masq://eth_t1.as45cs5c5:1.2.3.4:4444";
+    fn parse_works_for_ropsten() {
+        let descriptor = "masq://eth-ropsten:as45cs5c5@1.2.3.4:4444";
 
-        let result = NodeDescriptor::try_dismantle_str(descriptor).unwrap();
+        let result = NodeDescriptor::parse_url(descriptor).unwrap();
 
-        assert_eq!(
-            result,
-            (Blockchain::EthRopsten, "as45cs5c5", "1.2.3.4:4444")
-        )
+        assert_eq!(result, (Chain::EthRopsten, "as45cs5c5", "1.2.3.4:4444"))
     }
 
     #[test]
-    fn try_dismantle_str_works_for_ethereum_testnet_rinkeby() {
-        let descriptor = "masq://eth_t2.as45cs5c5:1.2.3.4:4444";
+    fn parse_works_for_dev_chain() {
+        let descriptor = "masq://dev:as45cs5c5@1.2.3.4:4444";
 
-        let result = NodeDescriptor::try_dismantle_str(descriptor).unwrap();
+        let result = NodeDescriptor::parse_url(descriptor).unwrap();
 
-        assert_eq!(
-            result,
-            (Blockchain::EthRinkeby, "as45cs5c5", "1.2.3.4:4444")
-        )
+        assert_eq!(result, (Chain::Dev, "as45cs5c5", "1.2.3.4:4444"))
     }
 
     #[test]
-    fn try_dismantle_str_works_for_dev_chain() {
-        let descriptor = "masq://dev.as45cs5c5:1.2.3.4:4444";
+    fn parse_works_for_polygon_mainnet() {
+        let descriptor = "masq://polygon-mainnet:as45cs5c5@1.2.3.4:4444";
 
-        let result = NodeDescriptor::try_dismantle_str(descriptor).unwrap();
+        let result = NodeDescriptor::parse_url(descriptor).unwrap();
 
-        assert_eq!(result, (Blockchain::Dev, "as45cs5c5", "1.2.3.4:4444"))
+        assert_eq!(result, (Chain::PolyMainnet, "as45cs5c5", "1.2.3.4:4444"))
     }
 
     #[test]
-    fn try_dismantle_str_complains_about_url_prefix_not_found() {
-        let descriptor = "https://eth.as45cs5c5:1.2.3.4:4444";
+    fn parse_works_for_mumbai() {
+        let descriptor = "masq://polygon-mumbai:as45cs5c5@1.2.3.4:4444";
```

Comment:
Is this 'polygon-mumbai' a segment that will be passed to Chain::from? It uses just 'mumbai' to match, correct?

Reply:
The same things as previously. I think this is okay. However I'm going to reflect your idea about having two valid syntax cases to call a chain excluding places where it's used within a descriptor.